queue.go

```diff
@@ -219,13 +219,18 @@ func (queue *redisQueue) StartConsuming(prefetchLimit int, pollDuration time.Dur
 	return true
 }
 
-func (queue *redisQueue) StopConsuming() *sync.WaitGroup {
+func (queue *redisQueue) StopConsuming() <-chan struct{} {
 	if queue.deliveryChan == nil || atomic.LoadInt32(&queue.consumingStopped) == int32(1) {
-		return &queue.stopWg // not consuming or already stopped
+		return nil // not consuming or already stopped
```

Comment:
I think we should probably return a closed channel here. Unless we feel that this is a pathological case, then we should panic().

Although this probably won't be a problem in practice I don't think the system for managing the state of the queue

1. Just Created
2. Consuming
3. Stopped

Is really thread-safe.

Reply:
updated. now returns a closed channel instead of nil

as discussed, we don't expect the queue be switched on and off multiple times through different go routines that may causing racing condition. so thread-safe for switching between those states is not critical at this point. we may consider it once we start to support it be on/off multiple times